crypto/keys/secp256k1/internal/secp256k1: secp256_test.go fails when CGO_ENABLED=0 yet other files are guarded and run only with cgo enabled

<!-- < < < < < < < < < < < < < < < < < < < < < < < < < < < < < < < < < ☺ 
v                            ✰  Thanks for opening an issue! ✰    
v    Before smashing the submit button please review the template.
v    Please also ensure that this is not a duplicate issue :)  
☺ > > > > > > > > > > > > > > > > > > > > > > > > > > > > > > > > >  -->

<!--
IMPORTANT: Prior to opening a bug report, check if it affects one of the core modules
and if its elegible for a bug bounty on `SECURITY.md`. Bugs that are not submitted
through the appropriate channels won't receive any bounty.
 -->

## Summary of Bug

Noticed when trying to add a static analyzer to catch resource leaks on every PR/push per https://github.com/cosmos/cosmos-sdk/pull/13246 that if you run tests with `CGO_ENABLED=0` there are numerous failures
```shell
$ CGO_ENABLED=0 go test ./crypto/keys/...
# github.com/cosmos/cosmos-sdk/crypto/keys/secp256k1/internal/secp256k1 [github.com/cosmos/cosmos-sdk/crypto/keys/secp256k1/internal/secp256k1.test]
crypto/keys/secp256k1/internal/secp256k1/secp256_test.go:66:14: undefined: Sign
crypto/keys/secp256k1/internal/secp256k1/secp256_test.go:89:12: undefined: Sign
crypto/keys/secp256k1/internal/secp256k1/secp256_test.go:91:12: undefined: RecoverPubkey
crypto/keys/secp256k1/internal/secp256k1/secp256_test.go:92:12: undefined: ErrInvalidRecoveryID
crypto/keys/secp256k1/internal/secp256k1/secp256_test.go:93:36: undefined: ErrInvalidRecoveryID
crypto/keys/secp256k1/internal/secp256k1/secp256_test.go:100:14: undefined: Sign
crypto/keys/secp256k1/internal/secp256k1/secp256_test.go:104:18: undefined: RecoverPubkey
crypto/keys/secp256k1/internal/secp256k1/secp256_test.go:118:15: undefined: Sign
crypto/keys/secp256k1/internal/secp256k1/secp256_test.go:122:15: undefined: Sign
crypto/keys/secp256k1/internal/secp256k1/secp256_test.go:151:15: undefined: Sign
crypto/keys/secp256k1/internal/secp256k1/secp256_test.go:151:15: too many errors
```

but if we look at the other files in that directory, we can see that they are guarded with cgo guards like in 
https://github.com/cosmos/cosmos-sdk/blob/b96bed5534d39e7ff67421ebd88df33ce8390783/crypto/keys/secp256k1/internal/secp256k1/panic_cb.go#L5-L6

https://github.com/cosmos/cosmos-sdk/blob/b96bed5534d39e7ff67421ebd88df33ce8390783/crypto/keys/secp256k1/internal/secp256k1/secp256.go#L5-L6

<!-- Concisely describe the issue -->

## Version

b96bed5534d39e7ff67421ebd88df33ce8390783

--- a/crypto/keys/secp256k1/internal/secp256k1/secp256_test.go
+++ b/crypto/keys/secp256k1/internal/secp256k1/secp256_test.go
@@ -2,6 +2,9 @@
 // Use of this source code is governed by a BSD-style license that can be found in
 // the LICENSE file.
 
+//go:build !gofuzz && cgo
+// +build !gofuzz,cgo
+
 package secp256k1
 
 import (

